src/components/sortable/helpers.ts

```diff
@@ -16,3 +16,23 @@ export const getPreviewStyles = (clientOffset: XYCoord | null): CSSProperties =>
     WebkitTransform: transform,
   };
 };
+
+export const calculateCursorBasedDropIndex = (
+  fromIndex: number,
+  targetIndex: number,
+  isTopZone: boolean,
+): number => {
+  if (isTopZone) {
+    // Dropping in top zone: insert BEFORE this item
+    // If dragging forward (fromIndex < targetIndex)
+    return fromIndex < targetIndex ? targetIndex - 1 : targetIndex;
```

Comment:
When possible try write code instead of writing comment
```suggestion
    const isDraggingForward = fromIndex < targetIndex;

    return isDraggingForward ? targetIndex - 1 : targetIndex;
```